backend/common/src/purl.rs

```diff
@@ -128,15 +128,18 @@ impl Debug for Purl {
 
 impl From<&str> for Purl {
     fn from(value: &str) -> Self {
-        PackageUrl::from_str(value).unwrap().into()
```

Comment:
@bobmcwhirter We're using a clippy rule that doesn't like `unwrap`. We only allow them for tests. I don't know if deriving `Default` is The Right Thing in this case or not, but seemed harmless.

Reply:
I think the answer is TryFrom instead since failure is an option.